src/components/QueryEditor.tsx

```diff
@@ -598,6 +596,123 @@ const QueryEditor: QueryEditorComponent = ({
       ).concat(DEFAULT_RETURN_NODE),
     [viewConditions]
   );
+  useEffect(() => {
+    // Create Initial Branch and Condition
+    if (view.uid === parentUid) return;
+
+    const branches = viewCondition?.conditions.length;
+    const branchUid = getNthChildUidByBlockUid({
+      blockUid: view.uid,
+      order: view.branch,
+    });
+    const branchConditions = getChildrenLengthByPageUid(branchUid);
+
+    if (!branches) createBranch();
+    if (!branchConditions && branches) {
+      addCondition({
+        parentUid: branchUid,
+        order: 0,
+        isNested: true,
+      });
+    }
+  }, [view, viewCondition]);
+  useEffect(() => {
+    // Navigate Branches with Left Or Right Arrows
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (view.uid === parentUid) return;
+      if ((event.target as HTMLElement).tagName === "INPUT") return;
+
+      const totalBranches = viewCondition?.conditions.length || 0;
+      const isFirstBranch = view.branch === 0;
+      const isLastBranch = view.branch - 1 === totalBranches;
+
+      if (event.key === "ArrowLeft" && !isFirstBranch) {
+        setViewStack(
+          viewStack.slice(0, -1).concat([
+            {
+              uid: view.uid,
+              branch: view.branch - 1,
+            },
+          ])
+        );
+        event.preventDefault();
+      }
+      if (event.key === "ArrowRight" && !isLastBranch) {
+        setViewStack(
+          viewStack
+            .slice(0, -1)
+            .concat([{ uid: view.uid, branch: view.branch + 1 }])
+        );
+        event.preventDefault();
+      }
+    };
+    document.addEventListener("keydown", handleKeyDown);
+    return () => document.removeEventListener("keydown", handleKeyDown);
+  }, [view, viewCondition]);
```

Comment:
[oos] In general, we should look for a way to look up component state without having them as dependencies to use effect that are truly meant to be set up once (on mount). One strategy for this is the ref approach above.